### ISSUE TITLE
Incrementing or Decrementing a value and then calling `get` returns _Map<String, dynamic>

### ISSUE THREAD

**Issue by Nidal-Bakir** [parse-community/Parse-SDK-Flutter#842]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/main/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest version of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse Flutter SDK](https://github.com/parse-community/Parse-SDK-Flutter/releases). 

### Issue Description
Incrementing or Decrementing a value using `setDecrement(key,amount)` or `setIncrement(key,amount)` and then calling `get(key)` returns `_Map<String, dynamic> `

related to: #834, #696

### Steps to reproduce

1. create `ParseObject`
2. increment or decrement a value using `setIncrement` or `setDecrement` functions
3. get the value using the value key with `get("key")`
4. the result of calling `get("key")` is `Map` which is not expected

code snippet:
```dart 
final dietPlansObject = ParseObject("Diet_Plans");

final anyAmount = 1;
dietPlansObject.setIncrement("key", anyAmount);
// dietPlansObject.setDecrement("key", anyAmount);

final value = dietPlansObject.get("key");

print(value.runtimeType); // _Map<String, dynamic>
``` 

### Actual Outcome
`_Map<String, dynamic>`

### Expected Outcome
`num`

### Environment

Parse Flutter SDK
- SDK version: `3.1.15`
- Operating system version: `any`

Server
- Parse Server version: `any`

### Logs
From tests error:

> Expected: <Instance of 'num'>
>   Actual: {'__op': 'Increment', 'amount': 1.0}
>    Which: is not an instance of 'num'
> package:test_api                                 expect
> test/src/objects/parse_object_test.dart 1180:11  main.<fn>.<fn>.<fn>

**Comment by mtrezza** [comment]
@Nidal-Bakir I'm removing you from all issues as assignee. We do not use the assignment feature.

**Comment by Nidal-Bakir** [comment]
Fix via #860
